File: app/src/main/java/ch/bailu/aat/services/tileremover/StateScanned.java
package ch.bailu.aat.services.tileremover;

import ch.bailu.aat.helpers.AppBroadcaster;

public class StateScanned implements State {
    private final StateMachine state;

    public StateScanned(StateMachine s) {
        state = s;
        AppBroadcaster.broadcast(state.context, AppBroadcaster.TILE_REMOVER_STOPPED);

    }


    @Override
    public void scan() {
        rescan();
    }


    @Override
    public void rescan() {
        state.set(new StateScanForRemoval(state));
    }

    @Override
    public void stop() {}

    @Override
    public void reset() {
        state.set(new StateScan(state));
    }

    @Override
    public void remove() {
        if (state.summaries.getRemoveCount() > 0) {
            state.set(new StateRemove(state));
        }
    }

}

File: aat-android/src/main/kotlin/ch/bailu/aat/services/tileremover/StateScanned.kt
package ch.bailu.aat.services.tileremover

import ch.bailu.aat_lib.dispatcher.AppBroadcaster

class StateScanned(private val state: StateMachine) : State {
    init {
        state.broadcast(AppBroadcaster.TILE_REMOVER_STOPPED)
    }

    override fun scan() {
        rescan()
    }

    override fun rescan() {
        state.set(StateScanForRemoval(state))
    }

    override fun stop() {}
    override fun reset() {
        state.set(StateUnscanned(state))
    }

    override fun remove() {
        if (state.summaries[0].countToRemove > 0) {
            state.set(StateRemove(state))
        }
    }

    override fun removeAll() {
        state.set(StateRemoveAll(state))
    }
}
